CLAUDE.md

```diff
@@ -245,30 +297,126 @@ Performance collectors follow a comprehensive testing pattern:
+   // Helper function pattern - consistent naming and structure
+   func createTestXCollector(t *testing.T, procContent string, sysFiles map[string]string) (*XCollector, string, string) {
        tmpDir := t.TempDir()
-       // Create mock files
-       return collector
+       procPath := filepath.Join(tmpDir, "proc")
+       sysPath := filepath.Join(tmpDir, "sys")
+       
+       // Setup mock files...
+       
+       config := performance.CollectionConfig{
+           HostProcPath: procPath,
+           HostSysPath:  sysPath,
+       }
+       collector, err := collectors.NewXCollector(logr.Discard(), config)
+       require.NoError(t, err)
+       
+       return collector, procPath, sysPath
    }
    
-   // Validation helper pattern
-   func validateResults(t *testing.T, actual, expected *Stats) {
-       assert.Equal(t, expected.Field, actual.Field)
+   // Collection and validation helper
+   func collectAndValidateX(t *testing.T, collector *XCollector, expectError bool, validate func(t *testing.T, result TypedResult)) {
+       result, err := collector.Collect(context.Background())
+       
+       if expectError {
+           assert.Error(t, err)
+           return
+       }
+       
+       require.NoError(t, err)
+       typedResult, ok := result.(TypedResult)
+       require.True(t, ok, "result should be TypedResult")
+       
+       if validate != nil {
+           validate(t, typedResult)
+       }
    }
+   
+   // Test data as constants
+   const validProcFile = `actual /proc file content here`
+   const malformedProcFile = `malformed content`
    ```
 
 4. **Testing Requirements**
-   - **Path validation**: Test absolute vs relative paths
+   - **Constructor tests**: Separate test function for constructor validation
+   - **Path validation**: Test absolute vs relative paths, empty paths
    - **File interdependency**: Test behavior when related files unavailable
-   - **Return type validation**: Explicit type assertions
-   - **Graceful degradation**: Document critical vs optional files
+   - **Return type validation**: Explicit type assertions with proper error messages
+   - **Graceful degradation**: Document and test critical vs optional files
+   - **Boundary conditions**: Test zero values, maximum values (e.g., max uint64)
+   - **Malformed input**: Test partial data, missing fields, corrupt formats
+   - **Special cases**: Test virtual interfaces, disabled devices, etc.
 
 5. **Testing Principles**
    - Don't test static properties (Name, RequiresRoot)
    - Focus on parsing logic and error handling
    - Use realistic test data from actual /proc files
-   - Test collectors in `collectors_test` package
+   - Test collectors in `collectors_test` package (external testing)
+   - Name test functions consistently: `TestXCollector_Constructor`, `TestXCollector_Collect`
+   - Use descriptive test names that explain the scenario
+   - Group related test scenarios in the same test function
```

Comment:
Also add don't add tests to check for interface validation. We have the compile time check for that